src/service.py

```diff
@@ -0,0 +1,75 @@
+"""Utility module to help manage the snap service with guarding functions."""
+
+from logging import getLogger
+from typing import Any, Optional
+
+from charms.operator_libs_linux.v2 import snap
+
+logger = getLogger(__name__)
+
+
+class SnapService:
+    """A class representing the a snap service (only support one service)."""
+
+    def __init__(self, client: snap.Snap, snap_service: str) -> None:
+        """Initialize the class."""
+        self.snap_client = client
+        self.snap_service = snap_service
+
+    def is_active(self) -> bool:
+        """Return True if the snap service is active."""
+        return self.snap_client.services.get(self.snap_service, {}).get("active", False)
+
+    def start(self) -> None:
+        """Start and enable the snap service."""
+        self.snap_client.start([self.snap_service], enable=True)
+
+    def stop(self) -> None:
+        """Stop and disable the snap service."""
+        self.snap_client.stop([self.snap_service], disable=True)
+
+    def configure(self, snap_config: dict[str, Any]) -> None:
+        """Configure the snap service."""
+        # Bail out or it will crash on self.snap_client.set() method
+        if not snap_config:
+            logger.warning("empty snap config: %s, skipping...", snap_config)
+            return
+
+        # Store is previous state, for later use.
+        previously_active = self.is_active()
+
+        self.snap_client.set(snap_config, typed=True)
+
+        # Changing snap configuration will also restart the snap service, so we
+        # need to explicitly preserve the state of the snap service.
```

Comment:
Yes, it's okay, because normally the snap service will be started on install. We can fix it here or in the snap. But maybe better in the snap, we can do something like "Error: cannot configure snap because it snap is not start" or something like that.

Reply:
We can address it later if the snap side is updated